### PR TITLE
Commands to list builds, branches, projects

### PR DESCRIPTION
**Pull request by benabineri**
# Description of change

`list`

## Checklist

- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [x] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: cmd/resim/commands/commands.go [hunk 1]
```diff
@@ -53,6 +54,14 @@ func Execute() error {
 	return rootCmd.Execute()
 }
 
+func OutputJson(data interface{}) {
+	o, err := json.MarshalIndent(data, "", "  ")
```
[the remark this answers is not in the document]
Comment (iainjwhiteside) [reply cmd/resim/commands/commands.go:58]: not a working suggestion as the stuff below needs changed, maybe

File: cmd/resim/commands/commands.go [hunk 2]
```diff
@@ -53,6 +54,14 @@ func Execute() error {
 	return rootCmd.Execute()
 }
 
+func OutputJson(data interface{}) {
+	o, err := json.MarshalIndent(data, "", " ")
```
[the remark this answers is not in the document]
Comment (benabineri) [reply cmd/resim/commands/commands.go]: Good catch! It mangles `&` at least

File: cmd/resim/commands/commands.go [hunk 3]
```diff
@@ -53,6 +54,14 @@ func Execute() error {
 	return rootCmd.Execute()
 }
 
+func OutputJson(data interface{}) {
```
[the remark this answers is not in the document]
Comment (benabineri) [reply cmd/resim/commands/commands.go:58]: 🙇

File: testing/end_to_end_test.go [hunk 4]
```diff
@@ -179,16 +179,6 @@ func (s *EndToEndTestSuite) SetupSuite() {
 		os.Exit(1)
 	}
 
-	// Validate the client credential environment variables are set:
-	if !viper.IsSet(ClientID) {
```
[the remark this answers is not in the document]
Comment (benabineri) [reply testing/end_to_end_test.go:183]: I am using the config file and the test works without setting those env vars, I suppose we could check auth before running the test but the test does that already... by failing

File: README.md [hunk 5]
```diff
@@ -67,6 +67,11 @@ We track issues and feature requests using [Github Issues](https://github.com/re
 
 You will need Go installed.
 
+### Building the client
```
[the remark this answers is not in the document]
Comment (benabineri) [reply README.md:70]: neater ty

File: cmd/resim/commands/project.go [hunk 6]
```diff
@@ -73,9 +81,39 @@ func init() {
...
+			})
+		if err != nil {
+			log.Fatal("failed to list projects:", err)
+		}
+		ValidateResponse(http.StatusOK, "failed to list projects", response.HTTPResponse)
+		if response.JSON200 == nil {
+			log.Fatal("empty response")
+		}
+		pageToken = response.JSON200.NextPageToken
+		allProjects = append(allProjects, *response.JSON200.Projects...)
```
[the remark this answers is not in the document]
Comment (benabineri) [reply cmd/resim/commands/project.go:108]: IDE autocomplete coming in with the slam dunk

File: cmd/resim/commands/commands.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (benabineri) [reply cmd/resim/commands/commands.go]: have used that method (and called OutputJson in logs)

File: testing/end_to_end_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (iainjwhiteside) [reply testing/end_to_end_test.go:183]: i think the config file sets the variables by viper magic, but it's fair that the test does fail... Think it's fine to leave them out then

**Pull request merged**